app/models/comment.php

```diff
@@ -22,6 +22,162 @@ class Comment extends AppModel
+        return $comments;
+  }
+
+  public static function get($id)
+  {
+      $db = DB::conn();
+
+      $row = $db->row('SELECT * FROM comment WHERE id = ?', array($id));
+
+      if (!$row) {
+          throw new RecordNotFoundException('no record found');
+      }
+
+      return new self($row);
+  }
+
+  public function delete()
+  {
+    try {
+      $db = DB::conn();
+
+      $db->begin();
+      $db->query('DELETE FROM comment WHERE id = ?', array($this->id));
+      $db->commit();
+    } catch (Exception $e) {
+      $db->rollback();
+    }
+  }
+
+  public function isUserComment()
+  {
+    return $this->user_id === $_SESSION['user_id'];
+  }
+
+  public function addLike()
+  {
+    try {
+      $db = DB::conn();
+
+      $db->begin();
+      $params = array(
+                  'comment_id' => $this->id, 
+                  'user_id' => $_SESSION['user_id'] 
+                  );
+      $db->insert('likes', $params);
+      $db->commit();
+    } catch (Exception $e) {
+      $db->rollback();
+    }   
+  }
+
+  public function removeLike()
+  {
+    try {
+      $db = DB::conn();
+
+      $db->begin();
+      $params = array(
+                  $this->id, 
+                  $_SESSION['user_id']
+                  );
+      $db->query('DELETE FROM likes WHERE
+                  comment_id = ? && user_id = ?', $params);
+      $db->commit();
+
+    } catch (Exception $e) {
+      $db->rollback();
+    }
+  }
+
+  public function isCommentLiked()
+  {
+    $db = DB::conn();
+
+    $params = array(
+                  $this->id, 
+                  $_SESSION['user_id']
+                  );
+    
+    $comment_liked = $db->row('SELECT * FROM likes WHERE 
+                              comment_id = ? && user_id = ?', $params);
+    
+    return !$comment_liked;
+  }
+
+  public function countLike()
+  {
+    $db = DB::conn();
+
+    $total_likes = $db->value('SELECT COUNT(*) FROM likes WHERE 
+                              comment_id =?', array($this->id));
+    
+    return $total_likes;
+  }
+
+  public static function getMostLiked()
+  {
+    $comment = array();
+    $db = DB::conn();
+
+    $rows = $db->rows("SELECT id, comment_id, user_id, COUNT(comment_id) 
+                        AS total_likes
+                        FROM likes GROUP BY comment_id 
+                        ORDER BY total_likes DESC LIMIT 10;");
+
+    foreach($rows as $row) {
+      $comment[] = new self($row);
+    }
+
+    return $comment;           
+  }
+
+  public function getThreadId()
+  {
+    $db = DB::conn();
+
+    $thread_id = $db->row('SELECT thread_id FROM 
+                            comment WHERE id = ?', array($this->comment_id));
+
+    return implode($thread_id);
+  }
+
+  public function getCommentSnippet()
+  {
+    $db = DB::conn();
+
+    $comment_body = $db->row('SELECT body FROM comment WHERE id = ?', 
+                              array($this->comment_id));
+
+    $comment_snippet = substr(implode($comment_body), 0, 30) . "..." ;
```

Comment:
Please avoid magic numbers, and declare these to const variables instead :bow: :dart: